kerastuner/applications/efficientnet.py

```diff
@@ -0,0 +1,158 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Hypertunable version of EfficientNet based on Keras.applications."""
+
+import tensorflow as tf
+import tensorflow.keras as keras
+
+from tensorflow.keras import layers
+from tensorflow.keras.applications import efficientnet
+from tensorflow.keras.layers.experimental import preprocessing
+
+from kerastuner.engine import hypermodel
+
+
+EFFICIENTNET_MODELS = {'B0': efficientnet.EfficientNetB0,
+                       'B1': efficientnet.EfficientNetB1,
+                       'B2': efficientnet.EfficientNetB2,
+                       'B3': efficientnet.EfficientNetB3,
+                       'B4': efficientnet.EfficientNetB4,
+                       'B5': efficientnet.EfficientNetB5,
+                       'B6': efficientnet.EfficientNetB6,
+                       'B7': efficientnet.EfficientNetB7}
+
+EFFICIENTNET_IMG_SIZE = {'B0': 224,
+                         'B1': 240,
+                         'B2': 260,
+                         'B3': 300,
+                         'B4': 380,
+                         'B5': 456,
+                         'B6': 528,
+                         'B7': 600}
+
+
+class HyperEfficientNet(hypermodel.HyperModel):
+    """An EfficientNet HyperModel.
+
+      # Arguments:
+
+        input_shape: shape tuple, e.g. `(256, 256, 3)`.
+              Input images will be resized if different from
+              the default input size of the version of 
+              efficientnet base model used.
+              One of `input_shape` or `input_tensor` must be
+              specified.
+        input_tensor: Keras tensor to use as image input for the model.
+              One of `input_shape` or `input_tensor` must be
+              specified.
+        classes: number of classes to classify images into.
+        augmentation_model: optional Model or HyperModel for image augmentation.
+        **kwargs: Additional keyword arguments that apply to all
+            HyperModels. See `kerastuner.HyperModel`.
+    """
+    def __init__(self,
+                 input_shape=None,
+                 input_tensor=None,
+                 classes=None,
+                 augmentation_model=None,
+                 **kwargs):
+        if isinstance(augmentation_model, (hypermodel.HyperModel,
+                                           keras.Model)):
+            self.augmentation_model = augmentation_model
+        elif augmentation_model is None:
```

Comment:
err, this check is not very useful here. How about:

```
if not  isinstance(augmentation_model, (hypermodel.HyperModel, keras.Model)):
  raise ValueError()
if not classes:
  raise ValueError()
if input_shape is None and input_tensor is None:
  raise ValueError()

self.input_shape = input_shape
self.input_tensor = input_tensor
self.classes = classes
self.augmentation_model = augmentation_model
```

Reply:
I would like allowing this to be None.  So how about
```
if augmentation_model and not isinstance(augmentation_model, (hypermodel.HyperModel, keras.Model))
```